analysis_notebooks/Geography Analysis.ipynb

```diff
@@ -1141,7 +1220,110 @@
+    "    # calculate mean % of images with cat_name label over different income deciles\n",
+    "    cat_ind = 0\n",
+    "    try:\n",
+    "        cat_ind = dataset.categories.index(cat_name)\n",
+    "    except:\n",
+    "        print(\"Please enter a valid category name from: {0}\".format(dataset.categories))\n",
+    "        return\n",
+    "        \n",
+    "\n",
+    "\n",
+    "    plt.tight_layout()\n",
+    "\n",
+    "    choro_data[region_colname] = choro_data[region_colname].astype(str)\n",
+    "    region_to_income = choro_data[[region_colname, data_colname]].set_index(region_colname).T.to_dict('records')[0]\n",
+    "    image_count = {}\n",
+    "    for region in region_tags:\n",
+    "        if region != 'out_of_boundary' and region in region_to_income and len(region_to_id[region]) > 250:\n",
+    "\n",
+    "            # rhs is # of images from region with at least 1 label of interest / # of images in region\n",
+    "            image_count[region] = region_tags[region][cat_ind] / len(region_to_id[region])\n",
+    "\n",
+    "    income = []\n",
+    "    label_count = []\n",
+    "    for region in image_count:\n",
+    "        income.append(region_to_income[region] / 1000)\n",
+    "        label_count.append(image_count[region]*100)\n",
+    "    ########################################\n",
+    "    #### quantization work\n",
+    "    num_bins = 10\n",
+    "    # create df for quantization work, using pd.qcut\n",
+    "    quantized_image_counts = pd.DataFrame({'income': income, 'count': label_count})\n",
+    "    labels = np.arange(1, 10+1)*10\n",
+    "    quantized_image_counts['quantiles_10'] = pd.qcut(quantized_image_counts['income'], q=num_bins, precision=0, labels = labels)\n",
+    "    # average image counts over quantized income bins:\n",
+    "    image_count = [] # overwrrite image_count (already stored in pandas df)\n",
+    "    for i in labels:\n",
+    "        image_count.append(np.mean(quantized_image_counts[quantized_image_counts['quantiles_10'] == i]['count']))\n",
+    "\n",
+    "    model = LinearRegression().fit(np.asarray(labels).reshape((-1, 1)), image_count)\n",
+    "\n",
+    "    labels = [str(int(i)) for i in labels]\n",
+    "    plt.bar(labels, image_count)\n",
+    "    # abline(model.intercept_, model.coef_)\n",
+    "    plt.xlabel(\"Income decile of ZIP code\", fontsize = 30)\n",
+    "    plt.ylabel(\"Mean % of Images with label\\n for ZIP codes in this decile\", fontsize =30)\n",
+    "    plt.title(\"{}\".format(dataset.categories[cat_ind].capitalize()), fontsize = 30)\n",
+    "    tksize = 19\n",
+    "    plt.xticks(fontsize= tksize) \n",
+    "    plt.yticks(fontsize= tksize)\n",
+    "    \n",
+    "    \n",
+    "#     plt.savefig('/Users/home/Desktop/img_regressions/{0}_quantized'.format(dataset.categories[cat_ind]), dpi=300, bbox_inches='tight')\n",
+    "# todo: add the save fig somewhere\n",
```

Comment:
todo?

Reply:
added save fig in accordance to summary pdf